api/app/models/base.py

```diff
@@ -70,16 +72,27 @@ def delete(self, db_session: "Session") -> Type[Self]:
 
     def update(self, db_session: "Session") -> Type[Self]:
         with db_session as session:
+            session.add(self)
             session.commit()
             session.refresh(self)
             return self
 
     @classmethod
-    def id_to_uid(cls, id: str) -> UUID:
+    def to_uid(cls, identifier: Union[str, UUID], prefix:Optional[str] = None) -> UUID:
```

Comment:
since this is on base, any class _can_ be used to get an id from a uuid for any other class by overloading the prefix. Normally you wont, ie
`User.to_uid('user-abcdef-ghijkl-mnopqrs-tuvwzyz')` will get you a UUID('abcdef-ghijkl-mnopqrs-tuvwzyz'), and any other prefix will error. 

but to avoid circular imports in the model, you can do Base.to_uid(val, prefix="user") and get the same result.